### ISSUE TITLE
Heatmap 'select' apply to column instead of cell

### ISSUE THREAD

**Issue by TienZhao** [apache/echarts#14230]
### Version
4.9.3

### Reproduction link
[https://jsfiddle.net/2j1bgshz/3/](https://jsfiddle.net/2j1bgshz/3/)

### Steps to reproduce
Click any cell with data inside.

### What is expected?
The cell is selected, with a default gray border displayed.

### What is actually happening?
Besides the clicked cell, all of those in the same column look as if they were selected. 

---
In the console.log output of 'selectchanged' event, there is an array named selected[0][dataIndex]. It changes after every event. It stores the indexes of all selected cells, instead of columns.
I expect the style of selected cells can change accordingly, however, it goes wild to the whole column.
I think it's a bug.
I tried solving it by putting every row's data in a different series, but that seems not really smart.
Looking forward to better solutions :)

**Comment by bscholer** [comment]
I'm also running into this issue, but didn't discover it until converting our whole application over to eCharts.

I agree that this is a bug though, since all other actions on series-heatmap are cell-scoped (e.g. emphasis/hover, tooltip). If this is intended functionality, documentation would be helpful.

**Comment by pissang** [comment]
The reason is data without name will be assigned with a default name from category axis. So cells on the same column will have same name. This design is mainly for the bar, line charts. But it may have issue in the heatmaps. 

The solution is assign a unique name for each cell data.

**Comment by bscholer** [comment]
@pissang Makes sense! That change was successful! Thanks for the help, and for such a wide-reaching charting library!

**Comment by fvales** [comment]
> @pissang Makes sense! That change was successful! Thanks for the help, and for such a wide-reaching charting library!

I am facing the same issue... Could you provide an example of how I can assign unique name for each cell data item. Thanks.

**Comment by rozsazoltan** [comment]
> I am facing the same issue... Could you provide an example of how I can assign unique name for each cell data item. Thanks.

I created a simple guide on Stack Overflow for solving the problem. Essentially, I implemented the [solution](https://github.com/apache/echarts/issues/14230#issuecomment-963756101) mentioned by @pissang and included the relevant part of the documentation alongside it.

**[Selecting a Single Cell Instead of a Column on a Heatmap Chart: A Step-by-Step](https://stackoverflow.com/questions/76546935/echarts-select-a-cell-in-a-heatmap/76598477#76598477) - Stack Overflow Answer**

**Comment by yaitskov** [comment]
I just hit this bug.

**Comment by kingyue737** [comment]
unstale
